<html><meta http-equiv="content-type" content="text/html;charset=iso-8859-1"><title>An Introduction To Sequence Comparison and Database Search<p>UPF, January 2008</p></title></head><BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000"><center><table border=10 cellpadding="20" width="580" nosave><tr> <td align=center bgcolor="#CC9933"><h2><font color=white>An Introduction To Sequence Comparison and Database Search<p>UPF, January 2008</p><p>Cedric Notredame</p></font></h2></td></tr></table><br><br><br><center><h1><font size="2">Send your Questions to: <a href="mailto:cedric.notredame@crg.es">cedric.notredame@crg.es</a></h1></font><br><br><table BGCOLOR=white Align=center width=70% Border=10><tr><td><center><p><Small>This Entire Course Was Automatically Generated Using BED, the Bioinformatics Exercise Database. BED is a freeware available on request <a href="mailto:cedric.notredame@europe.com?subject=BED">Cedric Notredame</a></small></p></center></td></tr></table><BR><BR><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr align=center BGCOLOR="white" ><td colspan=6><b><center>OBJECTIVES</b></center><p align="justify">This course is an <b>8</b> hours primer on sequence alignments. Its goal is to present an overview of the basic concepts of sequence alignments and some of their applications.  The first two hours will be dedicated to <b>molecular evolution</b>. We will focus on the implications of molecular evolution on sequence variation. We will use these concepts to define homology. We will then see how specific mathematical models (the substitution matrices) have been derived in order to quantify the evolutionary relationship between sequences.  The next two hours will be used to introduce the Needleman and Wunsch algorithm (Dynamic programming), a very basic algorithm that makes it possible to derive pairwise alignments from the sequences while using the substitution matrices. Over the following 2 hours, we will see how these pairwise alignment methods can be applied to database searches and we will develop the main concepts behind the BLAST algorithm. I will finally introduce the notion of multiple sequence alignment and show how a group of related sequences can be compared in order to infer common properties. We will then see the main principles behins two multiple sequence alignment package: ClustalW and T-Coffee. <br>
</p></td></tr></table><br><br><table BGCOLOR="#CC9966"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr align=center BGCOLOR="#CC9933" ><td><b><font size=+2 color=white>Date</font></b></td><td><b><font size=+2 color=white>Location</font></b></td><td><b><font size=+2 color=white>Session</font></b></td><td><b><font size=+2 color=white>Title</font></b></td><td><b><font size=+2 color=white>Links</font></b></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Thu. 15/01 08:40 </b></td><td>UPF</td><td>LECTURE</td><td><font size=+1><b>Evolution, Homology and Substition Models[2 hours]</b></font></td><td><a href=./lectures/4.1.pairwise_intro.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Mon. 19/01 08:40 </b></td><td>UPF</td><td>LECTURE</td><td><font size=+1><b>An Introduction to Dynamic Programming (1) [1 hour] </b></font></td><td><a href=./lectures/4.3.pairwise_dp.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Thu. 22/01 09:40 </b></td><td>UPF</td><td>LECTURE</td><td><font size=+1><b>An Introduction to Dynamic Programming (2) [1 hour] </b></font></td><td><a href=./lectures/4.3.pairwise_dp.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Fri. 23/01 08:40 </b></td><td>UPF</td><td>LECTURE</td><td><font size=+1><b>BLAST and Database Searches[2 hours] </b></font></td><td><a href=./lectures/4.2.pairwise_blast.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Mon. 26/01 08:40 </b></td><td>UPF</td><td>LECTURE</td><td><font size=+1><b>Multiple Sequence Alignments(1)[1 hours] </b></font></td><td><a href=./lectures/5.1.msa_introduction.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Mon. 27/01 08:40 </b></td><td>UPF</td><td>LECTURE</td><td><font size=+1><b>Multiple Sequence Alignments(2)[1 hours] </b></font></td><td><a href=./lectures/5.1.msa_introduction.ppt>L</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Wed. 04/01 11:10 </b></td><td>UPF</td><td>PRACTICAL</td><td><font size=+1><b>Group A: Database Searches </b></font></td><td><a href=./practicals/practical_1/index.html>P</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Wed. 04/01 15:10 </b></td><td>UPF</td><td>PRACTICAL</td><td><font size=+1><b>Group B: Database Searches </b></font></td><td><a href=./practicals/practical_2/index.html>P</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Thu. 05/01 15:10 </b></td><td>UPF</td><td>PRACTICAL</td><td><font size=+1><b>Group C: Database Searches </b></font></td><td><a href=./practicals/practical_3/index.html>P</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr></table><BR><BR> </body> </html>