<html>
 <head>
   <meta http-equiv="content-type" content="text/html;charset=iso-8859-1">
   <meta name="generator" content="Adobe GoLive 4">
   <title>Bioinformatics Exercises</title>
 </head>
 <BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000">
    <center>
       <table border=10 cellpadding="20" width="580" nosave>
	<tr>
	  <td align=center bgcolor="FFCCAA">
	      <h2>Multipe Sequence Alignments:<BR>ClustalW</h2>
	 </td>
	</tr>
      </table>
      <br><br><br>
            
      <table BGCOLOR="#FFFFFF" Align="center" cellpadding=30  leftmargin="30" marginwidth="30" width="70%"  border="5"> 
	<tr>
	  <td>
	  <p>You are given the following <a href="./file/1thm2_ref5.fasta">sequences</a>.</p>
	       <OL>
	       <LI><p>Generate a default clustalW alignment, using the interactive menu and the prompt.  How many files does clustalw produce</p></LI>
	       <LI><p>Obtain a list of the command line parameters (clustalw -options). Consult the online documentation at the ebi [<a href="http://www-igbmc.u-strasbg.fr/BioInfo/ClustalW/Top.html">DOC</a>]</p></LI>
	       <LI><p><b>Gap Penalties</b>  : change the gap opening penalties. Are the alignments different ?</p></LI>
	       <LI><p><b>Matrices </b>      : change the matrix.</p></LI>
	       <LI><p><b>Guide  Tree </b>   : invert the order in which the sequences 1thm and sepr_thesp are incorporated in the alignment</p></LI>
	       <LI><p><b>Phylogenic Tree</b>:Compute a phylogenetic tree using ClustalW and your alignment</p></LI>
	       <LI><p><b>Reference aln  </b>:Compare this alignment with its BaliBase reference <a href="./file/1thm2_ref5.ref_aln">ref</a>.</p></LI>
	       </OL>  
	  </td>
	</tr>
      </table>
 </body>
 </html>
