 <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>

    <title>T-RMSD Home Page</title>
    
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-W0ZCV3Q3DG"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-W0ZCV3Q3DG');
</script>   

  </head>
    <BODY TEXT="#000000" BGCOLOR=white LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000">
    <FONT FACE="Arial">
    <center>
   
 <table width="80%"       cellpadding=15 cellspacing=0 bgcolor="#FFEBCD" border=1 bordercolor="#8B4513">
	  <tr>
	    <td>
	      <table width="100%" cellpadding=15 cellspacing=0 cellborder=0 valign="TOP" border=0>
		  <tr bgcolor="#FFEBCD" >
		    <td colspan=1 bgcolor="#FFEBCD" width="5%"><br></td>
		    <td colspan=1 bgcolor="#FFEBCD" width="40%"><br></td>
		    <td colspan=1 bgcolor="#FFEBCD" width="5%"><br></td>
		    <td colspan=1 bgcolor="#FFEBCD" width="40%"><br></td>
		    <td colspan=1 bgcolor="#FFEBCD" width="5%"><br></td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFEBCD">
		      <center><table><tr><td align="center"><b> <FONT size="7" color="#8B4513">T-RMSD <br>A Fine-Grained Structure Clustering Method</font></b></td></tr></table></center>
		    </td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFE4C4">
		      <HR>
		      <center>
			<table width="100%" cellpadding=15 cellspacing=0 cellborder=0 valign=top>
			    <tr align=center>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="../homepage.html"><font color="#8B4513">HOME</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#SERVERS"><font color="#8B4513">SERVERS</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#DOWNLOAD"><font color="#8B4513">DOWNLOAD</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#COMMANDS"><font color="#8B4513">DOCUMENTATION</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#OUTPUT"><font color="#8B4513">OUTPUT</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#CONTACT"><font color="#8B4513">CONTACT</font></a></b></td>
			      
			    </tr>
			    
			</table>
		      </center>
		      <HR>
		    </td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  <tr>
		    <td rowspan=3 bgcolor="#FFEBCD"><br></td>
		    <td rowspan=3 bgcolor="white" valign="top">
		      <center><h2><FONT color="#8B4513">What is T-RMSD?</FONT></h2></center>
		      <table bgcolor="white">
			  <tr>
			    <td>
			      <P align="justify">T-RMSD (Tree based on Root Mean Square Deviation) is a distance RMSD (dRMSD) based method which generate Structural Trees (analogue to phylogenetic tree) to determine fine-grained structural variations associated with a given Multiple Sequence Alignment (generated by the user using the method of is choice). It is part of the T-Coffee distribution. The specificity of T-RMSD compared to other structural comparison methods stems from its capacity to generate a structural tree with values  equivalent bootstrap values supporting the structural clustering. Such clustering is achieved by the construction of matrixes of distances, calculated between equivalent residues as defined by the ungapped column of the given MSA. The resulting matrixes are then combined using the CONSENSE program from the PHYLIP package to generate a consensus structural tree with equivalent bootstrap values supporting each node(from 0 to 100; 100 indicating that all positions support the clustering).  </P>
			      <P>A list of related publications is available <a href="http://www.ncbi.nlm.nih.gov/pubmed/20471393,16873492,17032685" target="_blank" >here</a> and  <a href=./publications>there</a>. The following <a target="_blank" href="./presentations/trmsd.ppt">presentation</a> provides an overview of the algorithm and its validation.</p>
			    </td>
			  </tr>
		      </table>
		      
		    </td>
		    
		    <td colspan=1 rowspan=3 bgcolor="#FFEBCD"><br></td>
		    <td colspan=1 rowspan=1 bgcolor="white">
		      <center><h2><FONT color="#8B4513"><a name="SERVERS">SERVERS</a></font></h2></center>
		      <table bgcolor="white">
			
			  <tr><td>
			      <UL>
				<LI><A HREF="http://tcoffee.crg.cat/apps/tcoffee/do:trmsd">T-RMSD server</A></LI>
			      </UL>
			    </td></tr>
		      </table>
		    <td rowspan=3 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  <tr>
		    <td bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  <tr>
		    <td colspan=1 rowspan=1 bgcolor="white">
		      <center><h2><FONT color="#8B4513"><a name="DOWNLOAD">DOWNLOAD</a></font></h2></center>
		      <table bgcolor="white">
			  <tr><td>
			      <UL>
				<LI>T-RMSD is a special option allowing a fine-grained structural clustering of a given group of related proteins (i.e. protein domains). 
				   Download the latest T-Coffee version <a href="../tcoffee/index.html#DOWNLOAD">here</a>.</LI>
			      </UL>
			    </td></tr>
		      </table>
		    </td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  <tr>
		    <td bgcolor="#FFEBCD"><br></td>
		    <td colspan=3 bgcolor="white" valign="top">
		      <center><h2><FONT color="#8B4513"><a name="INSTALLATION">INSTALLATION</a></font></h2></center>
		      <table bgcolor="white">
			  <tr>
			    <td>
			      <P align="justify"><b>1-</b>Follow the standard T-Coffee <a href="../tcoffee/index.html#INSTALLATION">installation procedure</a>.</P>
			    </td>
			  </tr>
			  <tr>
			    <td>
			      <P align="justify"><b>2-</b>Install all the multiple sequence alignment packages you want to construct your Multiple Sequence Alignment. 
			      You should also take a look at the <a href="../tcoffee/index.html#DOCUMENTATION">T-Coffee Technical Documentation</a>. The latest list of packages interfaced with T-Coffee can also be obtained with the simple intruction
			      <code>  t_coffee > <a target="_blank" href="../tcoffee/package_list.txt">package_list</a> </code></P>
			    </td>
			  </tr>
		      </table>
		    </td>
		    <td bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  <tr>
		    <td colspan=1 rowspan=5 bgcolor="#FFEBCD"><br></td>
		    <td colspan=1 rowspan=5 bgcolor="white" valign="top">
		      <center><h2><FONT color="#8B4513"><a name="OUTPUT">OUTPUT</a></font></h2></center>
		      <center><table bgcolor="white">
			    <tr>
			      <td>
				<P align=justify>Given a Multiple Sequence Alignment <code>foo.aln</code> and a template file <code>foo.template_list</code> (similar to Expresso procedure; it corresponds to the association between sequences contained in the MSA and the PDB files used for the distances calculation) T-RMSD outputs a file named <code>foo.html</code> that contains a color coded indication of the ungapped column supporting the structural clustering. Gapped columns are not colored as they are not used to establish the clustering. Red brick columns indicate positions with a strong support to the clustering, while blue regions exhibit a low support. Red brick columns represent the most structurally discriminating positions. Unlike alginments generated using T-Coffee, there is no consistency score (0-100) for each sequence of the MSA. T-RMSD also outputs several files named <code>foo.struc_tree10, foo.struc_tree50, foo.struc_tree100</code> which corresponding to Neighbour Joining files, that can be used to build the structural tree using the software of your choice. The indication 10, 50 or 100 represent the pourcentage of ungapped columns used to generate the clustering.  </P>
				<a href="./images/CRD_tree.png" target="_blank" title="Click to zoom"><img border="0" src="./images/CRD_tree.png" width="350" height="350" ></a>
			      </td>
			    </tr>
			</table></center>
		    </td>
		    <td colspan=1 rowspan=5 bgcolor="#FFEBCD"><br></td>
		    <td colspan=1 bgcolor="white" valign="top">
		      <center><h2><FONT color="#8B4513"><a name="COMMANDS">DOCUMENTATION</a></font></h2></center>
		      <table bgcolor="white">
			  <tr>
			    <td>
			      <p>The full documentation is on the <a href="../tcoffee/index.html#DOCUMENTATION">T-Coffee Homepage</a>. But the following shortcuts may be useful.</p>
			  <TABLE border=0 WIDTH="100%" CELLPADDING="5" bgcolor="white">
			    <tr>
			      <TD bgcolor="grey"></td>
			    </tr>
			    <TR>
			      <TD>To run a T-RMSD from a given MSA, use the following</td>
			    </TR>
			    <tr>
			      <td>
				<code>t_coffee -other_pg trmsd <i>foo.aln</i> -template_file <i>foo.template_list</i> -mode msa2tree</code>
			      </td>
			    </TR>
			    <tr>
			      <TD bgcolor="grey"></td>
			    </tr>
			    <TD>Sample Datasets are available from: <a href="./data/dataset4trmsd.tar.gz">here</a></td>
			    <tr>
			      <TD colspan=3 bgcolor="grey"></td>
			    </tr>
			  </table>
			      
			    </td>
			  </tr>
		      </table>
		    </td>
		    <td colspan=1 rowspan=5 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  <tr>
		    <td colspan=1 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  <tr>
		    <td bgcolor="white" valign="top">
		      <center><h2><FONT color="#8B4513"><a name="LINKS">LINKS</a></font></h2></center>
		      <table bgcolor="white">
			  <tr>
			    <td>
			      <LI><A HREF="http://ekhidna.biocenter.helsinki.fi/dali_server/">Dali server</A> Dali server v3.0 </LI>
			    </td>
			  </tr>
			  <tr>
			    <td>
			      <LI><A HREF="http://www.cathdb.info/index.html">CATH database</A> CATH homepage v3.3 </LI>
			    </td>
			  </tr>
			  <tr>
			    <td>
			      <LI><A HREF="http://scop.mrc-lmb.cam.ac.uk/scop/">SCOP server</A> SCOP homepage 1.75 release </LI>
			    </td>
			  </tr>	  
		      </table>
		    </td>
		  </tr>
		  
		  <tr>
		    <td colspan=1 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  <tr>
		    <td bgcolor="white" valign="top">
		      <center><h2><FONT color="#8B4513"><a name="CONTACT">CONTACT</a></font></h2></center>
		      <table bgcolor="white">
			  <tr>
			    <td>
			      <P>Our projects relie on your feeback. Please send me an
				E-mail if you wish to make a request, a comment, or report a bug! 
				<code>
				  <br><br>
				  *******************************************<br>
				  Dr. Cedric Notredame, PhD.                 <br>       
				  Group Leader                               <br>
				  Comparative Bioinformatics Group           <br>
				  Bioinformatics and Genomics Programme      <br>
				  Center for Genomic Regulation (CRG)        <br>
				  Dr Aiguader, 88                            <br>
				  08003 Barcelona                            <br>
				  Spain                                      <br>
				  Email:  <A href="MAILTO:cedric.notredame@gmail.com">cedric.notredame@gmail.com</A>    <br>
				  HOME :  <A HREF="http://www.tcoffee.org/homepage.html">http://www.tcoffee.org/homepage.html</A><br>
				  GROUP:  <a href="http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics/03_Comparative_Bioinformatics">CRG</A><br>
				  Phone:  +34 933 160 271                <br>
				  *******************************************       <br>
				</code>
			    </td>
			  </tr>
		      </table>
		    </td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFEBCD"><br></td>
		  </tr>
		  
		  <tr>
		    <td colspan=5 bgcolor="#FFE4C4">
		      
		      <center>
			<HR>
				<table width="100%" cellpadding=15 cellspacing=0 cellborder=0 valign=top>
			    <tr align=center>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="../homepage.html"><font color="#8B4513">HOME</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#SERVERS"><font color="#8B4513">SERVERS</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#DOWNLOAD"><font color="#8B4513">DOWNLOAD</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#COMMANDS"><font color="#8B4513">DOCUMENTATION</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#OUTPUT"><font color="#8B4513">OUTPUT</font></a></b></td>
			      <td width="15%" onMouseOver="style.backgroundColor='white';"onmouseout="style.backgroundColor='#FFE4C4'"><b><a href="#CONTACT"><font color="#8B4513">CONTACT</font></a></b></td>
			      
			    </tr>
			    
			</table>
			<HR>
		      </center>
		      
		    </td>
    </center>
  </td>
  </tr>
  </table>
  </td>
  </tr>
  </table>
  </center>
  </font>
  </body>
</html>
