<HTML>
<HEAD>
   <META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
   <META NAME="GENERATOR" CONTENT="Mozilla/4.05 [en] (X11; I; IRIX 6.3 IP32) [Netscape]">
   <META NAME="Author" CONTENT="cedric Notredame">
   <META NAME="Keywords" CONTENT="Sociological data analysis">
   <TITLE>Pavie Matrix Project Home page</TITLE>
</HEAD>
<BODY TEXT="#000000" BGCOLOR="#FFEBCD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000">

<HR WIDTH="100%"></H1>

<H1>PAVIE PROJECT</H1>
<HR WIDTH="100%">

<P>This page centralizes all the code information relative to the PAVIE/Bioinformatics project, a collaboration between the PAVIE  (Jacques-Antoine Gauthier and Eric Widmer) and the Swiss Institute of Bioinformatics (Philipp Bucher, Cedric Notredame)</P>

<UL>
<LI><H2><FONT COLOR="#8B4513">Project #1: Matrix Training</FONT></H2></LI>
    <UL>
        <P>
      <LI><H3><FONT COLOR="#8B4513">Accessing and running the algorithms</FONT></H3></LI>
          <P><UL>
            <LI>Install the latest <a HREF="../../Packages/T-COFFEE_distribution.tar.gz">T-Coffee distribution </a> in which saltt is incorporated</LI>
            <LI>Use the <a href="./saltt_doc.doc">documentation</a> along with the following sample <a href=pavie_full.seq>sequences</a></LI>
          </UL></P> 

        <LI><H3><FONT COLOR="#8B4513">Training Matrices</FONT></H3></LI>
            <UL>
                <LI>Your sequences must be in FASTA format:<PRE>
            seq_reformat -in |<i>your sequences</i>| -action +pavie_seq2pavie_mat [_IDXX_TWEXX[THRid]_[CHANNELn]</PRE></LI>
                <LI>The program runs iteratively. It starts with the ID matrix and updates its matrix until convergeance is reached.</LI>
                <LI><b>Weight: _TWEXX_</b></LI>
                <UL>
                   <p>Various mode of identity measure are implemented (Trainning WEighting)</p>
                    <LI>_TWE00_: No weighting</LI>
                    <LI>_TWE01_: Default, n_id_pairs/n_match</LI>
                    <LI>_TWE02_: n_id_pairs/aln_length</LI>
                    <LI>_TWE03_: n_id_pairs/MIN(length seq1, length seq2)<b>Default</b></LI>
                    <LI>_TWE04_: n_id_pairs/MAX(length seq1, length seq2)<b>Default</b></LI>
                    <LI>_TWE05_: score as measured with the matrix<b></b></LI> 
                    <LI>simweight: <b>NOT SUPPORTED ANYMORE</b></LI>
                </UL>
                <LI><b>_THRid_</b></LI>
                <UL>
                    <LI>_THRid_: id is a threshold that filters alignments. For instance, with THR50, the program will only extract the counts from alignments with more than 50% ID or SIM (depending on the weighmode). The purpose of this filter is to discard bad alignments that should not contribute at all to the final matrix</LI>
               </UL>
               <LI><b>_SAMPLEn_</b></LI>
               <UL>
                    <LI>_SAMPLEn_: n is the number of random pairs sampled (without removal) at each training round. The pairs change at each round</LI>
                    <LI>_SAMPLE100_: will randomly select 100 pairs (different at each round)</LI>
                    <LI>_SAMPLE0_: will select all the pairs (Default)</LI> 
               </UL>
               
               <LI><b>Identity Measure: _IDX_</b></LI>
               <UL>
                    <p>Various mode of identity measure are implemented</p>
                    <LI>_ID01_: Default, n_id_pairs/n_match</LI>
                    <LI>_ID02_: n_id_pairs/aln_length</LI>
                    <LI>_ID03_: n_id_pairs/MIN(length seq1, length seq2)</LI>
                    <LI>_ID04_: n_id_pairs/MAX(length seq1, length seq2)</LI>
                    <LI>_ID05_: score as measured with the matrix<b>[Not implemented]</b></LI>
               </UL>
               <LI><b>_MCHSCOREn: Multichanel scoring scheme</b></LI>
               <UL>
                    <p>Various mode of identity measure are implemented</p>
                    <LI>MCHSCORE0: average over all the channels [Default]</LI>
                    <LI>MCHSCORE1: Minimum over all the channels</LI>
                    <LI>MCHSCORE2: Maximum over all the channels </LI>
                </UL>
               <LI><b>_CHANNELn_: multiple channel strings</b></LI>
               <UL>
                    <LI>_CHANNELn_: n is the number channels in the FASTA sequences. </LI>
                    <LI>_CHANNELn_: By default, n is set to 1</LI>
                    <LI><b>Principle for multiple channel matrix training</b>
	                <UL>
	                   <LI>One can use as many channels as required.</LI>
	                   <LI>Alphabets are totally independant (i.e. each channel can use the same alphabet)</LI>
	                   <LI>All the channels are simultaneouslz compared and one matrix per channel is estimated</LI>
	                   <LI>Using pavie_seq2pavie_aln, it is possible to use multiple channle matrices to align multiple channel strings</LI>
	               </UL>
	            <LI><b>Rules for multiple channel strings</b></LI>
                    <UL>
                        <LI>All the channels are in the same FASTA file</LI>
                        <LI>All the string of Channel X are grouped</LI>
                        <LI>All the channels must contains EXACTLY the same number of strings</LI>
                        <LI>String N of channel X CORRESPONDS to string N of channel Y</LI>
                        <LI>String N of channel X Must have the same lenght as string N of channel Y</LI>
                   </UL>
                   <LI><b>Adding the age in the multichanneling</b></LI>
                       <UL>
                           <LI>The age can be used as a channel</LI>
                           <LI>36 symbols (A-Z0-9) are available 
                   </UL>
                   <LI><b>Example of Multi-Channel FASTA file (Names are arbitrary): File: myseq.fasta</b>
                   <PRE>
                            >string1.channel1
                             abcdef
                            >string2.channel1
                             ab
                            >string3.channel1
                             abc
                            >string1.channel2
                             abzeff
                            >string2.channel2
                             ef
                            >string3.channel2
                             fxx
                         </PRE>
                        
               </UL>
               <LI><b>Examples of command Lines</b></LI>
               <UL>
                    <LI>Defaults: without any extra parameter THR0_SAMPLE0 is assumed. The weight mode must be specified</LI>
                    <LI><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_mat _SAMPLE10_TWE00_CHANNEL2_</LI>
                    <LI><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_mat _SAMPLE10_TWE00_</LI>
                    <LI><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_mat _TWE01_THR40_SAMPLE20_</LI>
                    <LI><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_mat _THR40_SAMPLE20_TWE02_</LI>
	            <LI><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_mat _THR40_SAMPLE20_TWE02_MCHSCORE1_</LI>
              </UL>  
              <LI>The trainning procedure outputs two series of file:</LI>
              <UL>
                  <LI>Matrix file     : <i>pavie_matrix.ch_#.cy_#.pavie_mat</i> that contain a matrix describing the specified chanel</LI>
                  <LI>Matrix list file: <i>pavie_matrix.cy_#.mat_list</i> that contain a list of matrix files</LI>
	          <LI>Matrix list file: can be used to compute alignments (next section)</LI>
              </UL>
      </UL>
      <LI><H3><FONT COLOR="#8B4513">Using the age as a channel</FONT></H3></LI>
      <UL>
          <P>
	  It is possible to use the age as a channel. This simply requires generating two extra channels that will be used to encode the age, along with the associated substitution matrices <P>
	  <UL>
	     <LI><PRE><b>EXAMPLE:</b>seq_reformat -in myseq.fasta -output pavie_age_channel -out xyz</PRE></LI>
      	     <LI>xyz_pavie_age_matrix.mat_list is a mat list that must be concatenated to the other mat list (cf multi channeling)</LI>
             <LI>xyz_pavie_age_matrix.mat_list is a mat list that must be concatenated to the other sequence file (cf multi-channeling)</LI>
             <LI>xyz_age_channel.fasta contains the sequences recoded (decades: channel1, years: channel2). These sequences must be concatenated to the other channels, as indicated in the channel section</LI>
             <LI><b>Note:</b> In the age_sequences, A=0...J=9</LI>
             <LI><b>Note:</b> It is possible to set the year corresponding to the first symbol of a sequence in the header:
                              <PRE>>name _FIRSTYEARXX_</PRE>
	                      where XX will be used as the offset of the first year</PRE>
	     <LI><b>Note:</b> Gaps are ignored<\LI> 
	 </UL></P>
	 
      </UL>
      <LI><H3><FONT COLOR="#8B4513">Validation of The Training Procedure</FONT></H3></LI>
      <UL>
          <P>
	  Validation is made by replacing a symbol (a for instance) with two other arbitrarily chosen symbols 
          (c and d) that are otherwise absent from the sequences. The substitution is made across the entire sequence set</P>
	  <P>
	  The new dataset should then be used to train a matrix. If the trainning procedure is adequate, the matrix should have the following properties:</P>
	  <p><UL>
	     <LI>cost[c][d]=cost[d][c] ~ 0</LI>
	     <LI>cost[c][x]~cost[d][x]</LI>
	     </UL></P>
	  <P>The random sequences can be generated as folows:</P>
	  <PRE><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2random_seq axw > outseq</PRE>
	  <P> in this case, axw indicates that a will be replaced with x OR w</P>
	</UL>
    <LI><H3><FONT COLOR="#8B4513">Clustering Sequences: Distance Matrix Computation</FONT></H3></LI>
            <UL>
               <LI>Given a trained matrix (previous bullet), it is possible to compute all the alignments 
	and the associated distance matrix betwween all the sequences (SA format). It is also possible to 
	specify the measure that will be used to output the distances.</LI>
               <LI>Parameters
               <UL>
                  <LI>_MATDIST_: a distance   matrix, where the distance measure mode can be specified with _IDX_</LI>
                  <LI>_MATSIM_ : a similarity matrix, where the distance measure mode can be specified with _IDX_</LI>
                  <LI>_IDXX_: the default is _ID01_</LI>
               </UL></LI>
               <LI>Example
               <PRE><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_aln pavie_matrix.cycle_1.mat_list _MATDIST_ID01_</PRE></LI>
               <LI>matrix_list</LI> 
	           <UL>
	           <LI>file containing the name of valid matrix files.</LI> 
                   <LI>The number of matrices defines the number of channel in the sequences (cf CHANNEL option).</LI>
                   <LI>A matrix_list file is output automatically when trainning matrices.</LI>
                   </UL>
                </LI>
            </UL>  
	  

      <LI><H3><FONT COLOR="#8B4513">Computing Alignments</FONT></H3></LI>
            <UL>
                <LI>Given a trained matrix (previous bullet), it is possible to compute all the alignments:
                <PRE>EXAMPLE: seq_reformat -in myseq -action +pavie_seq2pavie_aln pavie_matrix_.cy_0.mat_list |</PRE></LI>
                <LI>matrix_list
	           <UL>
	              <LI>file containing the name of valid matrix files.</LI> 
                      <LI>The number of matrices defines the number of channel in the sequences (cf CHANNEL option).</LI>
                      <LI>A matrix_list file is output automatically when trainning matrices.</LI>
                   </UL>
                </LI>
                <LI>Example
	        <PRE><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_aln pavie_matrix.cycle_1.mat_list _ID02_</PRE></LI>
                <PRE><b>EXAMPLE:</b> seq_reformat -in myseq.fasta -action +pavie_seq2pavie_aln pavie_matrix.cycle_1.mat_list _ID02_MCHSCORE1_</PRE></LI>
             </UL>

       <LI><H3><FONT COLOR="#8B4513">Transition Matrices</FONT></H3></LI>
            <UL>
                <LI>Given a series of sequences, one can compute the log odd associated with transitions:
                <PRE><b>EXAMPLE:</b>seq_reformat -in myseq.fasta -output transitions</i>|</PRE></LI>
            </UL>

        <LI><H3><FONT COLOR="#8B4513">Tutorial and examples</FONT></H3></LI>
        <LI><H3><FONT COLOR="#8B4513">Related Links</FONT></H3></LI>
        <LI><H3><FONT COLOR="#8B4513">Contact</FONT></H3></LI>
        Email   :  <A MAILTO="cedric.notredame@europe.com">cedric.notredame@europe.com</A>    <br>
 
      </UL>
       
</BODY>
</HTML>
