<html><meta http-equiv="content-type" content="text/html;charset=iso-8859-1"><title>An Introduction To Sequence Comparison and Database Search<p>UAB, 2015</p></title></head><BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000"><center><table border=10 cellpadding="20" width="580" nosave><tr> <td align=center bgcolor="#CC9933"><h2><font color=white>An Introduction To Sequence Comparison and Database Search<p>UAB, 2015</p><p>Cedric Notredame</p></font></h2></td></tr></table><br><br><br><center><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>OBJECTIVES</b></center><p align="justify">This course is an <b>4</b> hours primer on sequence alignments. Its goal is to present an overview of the basic concepts of sequence alignments and some of their applications.  The first two hours will be dedicated to <b>molecular evolution</b>. We will focus on the implications of molecular evolution on sequence variation. We will use these concepts to define homology. We will then see how specific mathematical models (the substitution matrices) have been derived in order to quantify the evolutionary relationship between sequences.  The next two hours will be used to introduce the Needleman and Wunsch algorithm (Dynamic programming), a very basic algorithm that makes it possible to derive pairwise alignments from the sequences while using the substitution matrices. <br>
</p></td></tr></table><br><br><h1><font size="2">Send your Questions to: <a href="mailto:cedric.notredame@crg.eu">cedric.notredame@crg.eu</a></h1></font><br><br><table BGCOLOR="#CC9966"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr align=center BGCOLOR="#CC9933" ><td><b><font size=+2 color=white>Date</font></b></td><td><b><font size=+2 color=white>Location</font></b></td><td><b><font size=+2 color=white>Session</font></b></td><td><b><font size=+2 color=white>Title</font></b></td><td><b><font size=+2 color=white>Links</font></b></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>1</b></td><td>UAB</td><td>LECTURE</td><td><font size=+1><b>Evolution and Comparison Models</b></font></td><td><a href=./lectures/4.1.pairwise_intro.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>2</b></td><td>UAB</td><td>LECTURE</td><td><font size=+1><b>Dynamic Programming Based Sequence Comparison</b></font></td><td><a href=./lectures/4.3.pairwise_dp.ppt>L</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>3</b></td><td>UAB</td><td>PRACTICAL</td><td><font size=+1><b>Dynamic Programming</b></font></td><td><a href=./practicals/practical_1/index.html>P</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>4</b></td><td>UAB</td><td>LECTURE</td><td><font size=+1><b>Multiple Sequence Alignments: Algorithms and Strategies</b></font></td><td><a href=./lectures/5.1.msa_introduction.ppt>L</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr></table><BR><BR><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>REFERENCES</b></center><p align="justify"><p><align>1.        Claverie and Notredame, Bioinformatics for Dummies, 2007, Wiley
</p></align><p><align>2.        Durbin et al., Biological Sequence Analysis, 1999, Oxford Press
</p></align><p><align>3.        Patthy, Protein Evolution, 2007, Wiley
</p></align></p></td></tr></table><br><br><table BGCOLOR=white Align=center width=70% Border=10><tr><td><center><p><Small>This Entire Course Was Automatically Generated Using BED, the Bioinformatics Exercise Database. BED is a freeware available on request <a href="mailto:cedric.notredame@europe.com?subject=BED">Cedric Notredame</a></small></p></center></td></tr></table><BR><BR> </body> </html>