<html>
 <head>
  <meta http-equiv="content-type" content="text/html;charset=iso-8859-1">
  <meta name="generator" content="Adobe GoLive 4">
  <title>Bioinformatics Exercises</title>
 </head>
 <BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000">
    <center>
      <table border=10 cellpadding="20" width="580" nosave>
	<tr>
	  <td align=center bgcolor="FFCCAA">
	    <h2>Multipe Sequence Alignments:<BR>Orthology and Paralogy</h2>
	 </td>
	</tr>
      </table>
      <br><br><br>
            
      <table BGCOLOR="#FFFFFF" Align="center" cellpadding=30  leftmargin="30" marginwidth="30" width="70%"  border="5"> 
	<tr>
	  <td>
	    <OL>
		<LI><p>produce the multiple alignment and the phylogenic tree of the three following sequences in Swissprot.</p></li> 
	    <UL>
	      <LI>-SCG1_HUMAN</LI> 
	      <LI>-SCGA_XENLA</LI> 
	      <LI>-STHM_MOUSE</LI> 
	    </UL><BR>
	    <LI>Discuss the results. What is wrong. Propose a strategy?<BR> </LI>
	    <LI>Add sequences from Swiss_prot in your list, make a new multiple sequence alignment and a tree (make sure to include the SCG1 sequences from rat, human and mouse)<BR> </LI>
	    <LI>Discuss the results and propose a new strategy in order to obtain a better tree. <BR></LI>
	    <LI><b>Note:</b> 
	      <UL>
		<LI>To compute the tree, use <a href="http://www.fr.embnet.org/services/analyseq/cgi-bin/phylomat_in.pl">infobiogene</a>.</LI>
		<LI>To plot a tree, you may use <a href="  http://www.es.embnet.org/Doc/phylodendron/treeprint-form.html">phylodendron.</a></LI>
		<LI><b>Never use a clustalw guide tree file (.dnd) in place of a phylogenetic tree.</b></LI>
	  </OL>
	  </td>
	</tr>
      </table>
      <p><font size="2">Questions should be sent to <a href="mailto:cedric.notredame@igs.cnrs-mrs.fr">C.Notredame</a> </font></center>
 </body>
 </html>
