<HTML>
<HEAD>
   <META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
   <META NAME="GENERATOR" CONTENT="Mozilla/4.04 [en] (X11; I; IRIX 5.3 IP22) [Netscape]">
</HEAD>
<BODY>

<CENTER>
<H1>
Solution For The Exercises on The Alignment of Two Sequences</H1></CENTER>

<H2>
<A NAME="genetic code"></A>1.</H2>

<UL>1-The probability is function of the number of existing codons. For
instance, Leucine is six times more likely to happen by chance than tryptophan.

<P>2-Tryptophan and Methionine are only represented by one codon both.
This means that any one point mutation will turn them into something else.
In contrast, Valine is insensitive to a mutation on the third position.

<P>3-The main diagonal on a substitution matrix is an indicator of the
rate of mutation observed for an amino-acid. A high value indicates that
a residue is likely to remain unchanged throught evolution. Tryptophan
(W) is traditionnally the most conserved residue. This is a constrast with
the genetic code.
<BR>This means that the selection of mutation follows rules that are not
only statistical. For instance, the substitution of a tryptohan by some
other residue is often very destabilizing for a protein structure. As a
consequence, even if that type of mutation is frequent, occurences were
the mutation gets selected are very rare.

<P>[<A HREF="two_seq.html#genetic code">BACK</A>]</UL>


<P>&nbsp;
<H2>
<A NAME="two_seq_solutions_4"></A>4.</H2>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 1.The big square spot is a low-complexity
region, as it happens a serine-rich
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; region.
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 2.It is a mature mRNA plotted
against the corresponding gene. The
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; interruptions
in the diagonal are introns.
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 3.It's a lab artifact: during
the reverse transcription of an mRNA (an EST, in
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; that case),
the reverse transcriptase did a U-turn, so that the end of the
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; cDNA is the
reverse complement of its start (this is why you see a second
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; diagonal). This
is an example of why you should be careful with EST data.
<BR><A HREF="two_seq.html#two_seq_4">[BACK]</A>
<BR>&nbsp;
<H2>
<A NAME="two_seq_solutions_5"></A>5.</H2>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 1-This sequence contains nine
repeats, all homologous to one another, hence the stripes observed on the
plot.
<BR>&nbsp;
<BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 2-Lalign produces by default
the 10 best non-overlapping alignments. None of them gives you the coordinates
of the repeats, but by extrapolation it is possible to find a near exact
solution of:
<UL>
<UL>
<UL>
<UL>
<UL>-9 repeats with a lentgth of ~40 residues:
<UL>13-42, 43-72, 73-104,
<BR>105-134, 135-161,
<BR>162-191, 192-220, 221-251,
<BR>252-282</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
<A HREF="two_seq.html#two_seq_5">[BACK]</A>
</BODY>
</HTML>
