<html><meta http-equiv="content-type" content="text/html;charset=iso-8859-1"><title>RNA Analysis<p>Bologna University, 2013</p></title></head><BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000"><center><table border=10 cellpadding="20" width="580" nosave><tr> <td align=center bgcolor="#CC9933"><h2><font color=white>RNA Analysis<p>Bologna University, 2014</p><p>Cedric Notredame</p></font></h2></td></tr></table><br><br><br><center><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>OBJECTIVES</b></center><p align="justify">
In this course, I will start by introducing mainstream next generation sequencing methods. I will then discuss how these methods can be used to sequence, or re-sequence large genomes. I will then introduce RNA-Seq and the systematic sequencing of cell transcriptomes, along with the many challenges it entails, such as gene modeling and isoform quantification. I will then introduce the notion of multiple genome alignment and review existing methods, including some developed in our group. The course will then be focused on Long Non Coding RNAs. I will introduce the latest ENCODE results on this new class of transcripts and present the challenges of homology based annotation for Long Non Coding RNA. Some methods available for this task will described, including including the pipeline we developed for the ENCODE companion paper on Long Non Coding RNAs. We will see how this pipeline, and similar tools, can be deployed to produce homology-based annotation of newly sequenced genomes. The last part of the course will be dedicated to the computation of RNA multiple sequence alignment methods. I will first introduce the standard multiple sequence alignment method and show how these methods can be applied to nucleotide sequences. I will finish this series of lectures by showing how multiple sequence alignment methods can be adapted to deal with RNA secondary structure analysis, and how they can possibly be used to combine structural and sequence information.
</p></td></tr></table><br><br><h1><font size="2">Send your Questions to: <a href="mailto:cedric.notredame@crg.es">cedric.notredame@crg.es</a></h1></font><br><br><table BGCOLOR="#CC9966"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr align=center BGCOLOR="#CC9933" ><td><b><font size=+2 color=white>Date</font></b></td><td><b><font size=+2 color=white>Location</font></b></td><td><b><font size=+2 color=white>Session</font></b></td><td><b><font size=+2 color=white>Title</font></b></td><td><b><font size=+2 color=white>Links</font></b></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Next Generation Sequencing</b></font></td><td><a href=./lectures/9.1.lecture.nextGenerationSequencing.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>RNASeq</b></font></td><td><a href=./lectures/9.2.lecture.RNASeq.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Short Reads Mapping</b></font></td><td><a href=./lectures/4.4.lecture.short_read_mapping_algo.ppt>L</a></td></tr><tr align=center BGCOLOR="#FFFFCC" ><td><b>  </b></td><td>Bologna</td><td>BIBLIO</td><td><font size=+1><b>Short Read Mapping</b></font></td><td><a href=./biblio/4.4.paper.gem.pdf>P</a><a href=./biblio/4.4.paper.StringMatchingBWT_ReadThroughTranscription.pdf>P</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Pairwise Dynamic Programming</b></font></td><td><a href=./lectures/4.3.pairwise_dp.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Secondary Structure Prediction</b></font></td><td><a href=./lectures/4.5.lecture.secondary_structure_prediction.ppt>L</a></td></tr><tr align=center BGCOLOR="#FFFFCC" ><td><b> </b></td><td>Bologna</td><td>BIBLIO</td><td><font size=+1><b>Secondary Structure Prediction</b></font></td><td><a href=./biblio/4.5.paper.Eddy_sec_struc_primer.pdf>P</a><a href=./biblio/4.5.paper.RivasEddy_pseudo_knots.pdf>P</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Blast Algorithm</b></font></td><td><a href=./lectures/4.2.pairwise_blast.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Searching Non Coding RNAs</b></font></td><td><a href=./lectures/4.6.lecture.finding_ncRNA.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Multiple Sequence Alignment Algorithms</b></font></td><td><a href=./lectures/5.1.msa_introduction.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>The T-Coffee Algorithm</b></font></td><td><a href=./lectures/5.3.2.T-Coffee.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> </b></td><td>Bologna</td><td>LECTURE</td><td><font size=+1><b>Multiply Aligning Long Non Coding RNAs.</b></font></td><td><a href=./lectures/5.7.RNA_MSA.ppt>L</a></td></tr></table><BR><BR><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>REFERENCES</b></center><p align="justify"><p><align>1.        Durbin et al., Biological Sequence Analysis, 1999, Oxford Press
</p></align></p></td></tr></table><br><br><table BGCOLOR=white Align=center width=70% Border=10><tr><td><center><p><Small>This Entire Course Was Automatically Generated Using BED, the Bioinformatics Exercise Database. BED is a freeware available on request <a href="mailto:cedric.notredame@europe.com?subject=BED">Cedric Notredame</a></small></p></center></td></tr></table><BR><BR> </body> </html>