<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<HTML>
<HEAD>
   <META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
   <META HTTP-EQUIV="content-type" CONTENT="text/html;charset=iso-8859-1">
   <META NAME="generator" CONTENT="Adobe GoLive 4">
   <META NAME="Template" CONTENT="C:\PROGRAM FILES\MICROSOFT OFFICE\OFFICE\html.dot">
   <META NAME="Author" CONTENT="Victor Jongeneel">
   <META NAME="GENERATOR" CONTENT="Mozilla/4.08 [en] (X11; I; Linux 2.0.32 i686) [Netscape]">
   <TITLE>Day 2 morning</TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" LINK="#0000FF" VLINK="#800080">

<H1>
Databases on the Internet</H1>
Launch a browser on the local machine and try to find answers to the following
questions.
<BR>Good luck in cyberspace!
<H4>
1-1 Using ExPASy server <TT><A HREF="http://www.expasy.ch">www.expasy.ch</A></TT></H4>
a. Look for the amino acid sequence of <FONT COLOR="#9932CC">human carbonic
anhydrase 2</FONT>
<BR>b. Get the corresponding nucleic acid entries in EMBL and GenBank
<BR>c. Get a PDB file
<BR>d. Have a look at the chromosomal localisation of human carbonic anhydrase
2 using MIM or GDB
<BR>e. Is there a maize carbonic anhydrase? or a drosophila carbonic anhydrase?
<H4>
1-2</H4>
a. Look for SWISS-PROT entry of the yeast gene <FONT COLOR="#9932CC">RPL36B</FONT>
<BR>b. Follow the link to SGD (Saccharomyces Genome Database) and find
the chromosomal localisation
<BR>c. Get the SGD entry of the 2nd gene in 5' on the same chromosomal
strand
<BR>d. Follow the link to YPD
<BR>e. Find the subcellular localisation of the protein
<BR>f. Follow the link to SWISS-PROT
<BR>g. Have a look at the domain structure in PRODOM and get the list of
proteins with at least one common domain
<H4>
1-3</H4>
a. Find on the web a database for <FONT COLOR="#9932CC">CFTR</FONT> mutations
<H4>
1-4</H4>
a. Find the gene <FONT COLOR="#9932CC">teashirt</FONT> in FlyBase
<BR>b. Get the corresponding SWISSPROT, EMBL and TRANSFAC entries from
FlyBase
<H4>
1-5</H4>
a. Go to the <FONT COLOR="#9932CC">Genome Net</FONT> server in Japan
<BR>b. Find a database called <FONT COLOR="#9932CC">KEGG</FONT>
<BR>c. In KEGG, find the enzyme number <FONT COLOR="#9932CC">EC 1.2.3.4</FONT>
<BR>d. Get from KEGG the ENZYME entry in ExPASy, then from ENZYME the SWISS-PROT
entry, then from SWISS-PROT the corresponding PROSITE entry.
<BR>e. Have a look at <FONT COLOR="#9932CC">BRENDA</FONT> database from
ENZYME entry.
<P><A HREF="day2a_answers.html">[Answers]</A>
<H4>
2-1</H4>
a. Use SRS at <A HREF="http://ludwig-sun1.unil.ch:6001/">EMBnet.ch</A>
or at <A HREF="http://srs.ebi.ac.uk">EBI</A> or use ENTREZ at <A HREF="http://www.ncbi.nlm.nih.gov/Entrez/">NCBI</A>
and search for <FONT COLOR="#9932CC">IL-2 receptor</FONT> sequences in
EMBL. Can you find all with a single search (there are three chains in
the human and mouse receptors) ? How many entries in EMBL contain IL-2
receptor sequences?
<BR>b. Find the sequences of the mRNAs for the human alpha and gamma chains.
<BR>c. Follow cross-references to Swiss-Prot and Medline.
<BR>d. Retrieve Medline articles relevant to human IL-2 receptor sequences.
<BR>e. Retrieve the protein sequences directly from the Swiss-Prot server
on ExPASy (<A HREF="http://www.expasy.ch/">www.expasy.ch</A>). Is the information
different from the Swiss-Prot entries stored at the EBI? Try the Java feature
table viewer!
<H4>
2-2</H4>
a. Are the human IL-2R alpha and gamma genes present in the <A HREF="http://www.ncbi.nlm.nih.gov/Omim/">OMIM</A>
database?&nbsp; Where are they localized (chromosome, arm, band) ?
<BR>b. Can you find single sources containing links to all of the information
relevant to one gene?
<BR>c. View the cytogenetic maps of the regions surrounding the gene loci
(use the Mapview2 Java viewer).
<BR>d. Are known diseases assosiated with these genes?
<BR>e. What are the clinical synopses?
<BR>&nbsp;
<BR>&nbsp;
<P><A HREF="day2a_answers.html">[Answers]</A>

