<html><meta http-equiv="content-type" content="text/html;charset=iso-8859-1"><title>An Introduction To Sequence Comparison and Database Search<p>Bergen University, May 2020</p></title></head><BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000"><center><table border=10 cellpadding="20" width="580" nosave><tr> <td align=center bgcolor="#CC9933"><h2><font color=white>An Introduction To Sequence Comparison and Database Search<p>Bergen University, May 2020</p><p>Cedric Notredame</p></font></h2></td></tr></table><br><br><br><center><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>OBJECTIVES</b></center><p align="justify">The first part is dedicated to molecular evolution. We will focus on the implications of molecular evolution on sequence variation. We will use these concepts to define homology. We will then see how specific mathematical models (the substitution matrices) have been derived in order to quantify the evolutionary relationship between sequences. In the second part we will cover the pairwise and single sequence analysis. We will introduce the Needleman and Wunsch algorithm (Dynamic programming), a very basic algorithm that makes it possible to derive pairwise alignments from the sequences while using the substitution matrices. Implementation aspects will be extensively covered, including Linear Space implementations and pair-HMM alternatives for pairwise sequence alignments. These dynamic programming algorithms will be compared for both scope and application to the Burrow Wheeler Transform algorithm an essential component of genomic data aligners. The course will finish with an introduction of the Zuker RNA fold algorithm including an overview of the CYK algorithm . The third part of the course will deal with Multiple Sequence Alignment algorithms of both RNA and Protein sequences. We will introduce the basic notion of this important class of algorithm and will then move to the most recent algorithms, able to take into account protein and RNA 3D structure while building the models. We will discuss some recent results on the relationship between MSA computation and Tree reconstruction. The course will finish with a detailed description of the most recent large scale algorithm including the recently published Regressive Algorithm.
</p></td></tr></table><br><br><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>REQUISITES</b></center><p align="justify">Practicals will be in python 3. They will involve adapting existing scripts rather than programming from scratch. For this reason, students with no practical knowledge of python but with a good grasp of programming languages like Perl, C or JAVA should manage reasonnably well. Students are expected to bring their own lap-top and advised to have a LINUX boot. 
</p></td></tr></table><br><br><h1><font size="2">Send your Questions to: <a href="mailto:cedric.notredame@crg.eu">cedric.notredame@crg.eu</a></h1></font><br><br><table BGCOLOR="#CC9966"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr align=center BGCOLOR="#CC9933" ><td><b><font size=+2 color=white>Date</font></b></td><td><b><font size=+2 color=white>Location</font></b></td><td><b><font size=+2 color=white>Session</font></b></td><td><b><font size=+2 color=white>Title</font></b></td><td><b><font size=+2 color=white>Links</font></b></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Day 1</b></td><td>NORBIS</td><td>LECTURE</td><td><font size=+1><b>Pairwise comparisons in an evolutionary context</b></font></td><td> <a href=https://www.dropbox.com/s/jl2411kijh9vj1w/0.1.books.durbin.ppt?dl=0>Slides</a> <a href=https://www.dropbox.com/s/1rmlhc9mitfwbx3/4.1.1.Introduction.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 1</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Parsing Biological Files</b></font></td><td><a href=./practicals/practicals_1/index.html>P</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 1</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Computing Substitution Matrices</b></font></td><td><a href=./practicals/practicals_2/index.html>P</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Day 2</b></td><td>NORBIS</td><td>LECTURE</td><td><font size=+1><b>Introduction to Dynamic Programming</b></font></td><td> <a href=https://www.dropbox.com/s/dwk39nv77v43ba7/4.3.pairwise_dp.ppt?dl=0>Slides</a> <a href=https://www.dropbox.com/s/69ghu9woa3o6wml/4.3.pairwise_dp.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 2</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Introduction to Dynamic Programming 1</b></font></td><td><a href=./practicals/practicals_3/index.html>P</a> <a href=https://www.dropbox.com/s/xy0j9bqpwafzwri/ex.7.2P3.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 2</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Introduction to Dynamic Programming 2</b></font></td><td><a href=./practicals/practicals_4/index.html>P</a> <a href=https://www.dropbox.com/s/xy0j9bqpwafzwri/ex.7.2P3.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Day 3</b></td><td>NORBIS</td><td>LECTURE</td><td><font size=+1><b>Introduction to Multiple Sequence Alignment</b></font></td><td> <a href=https://www.dropbox.com/s/w571l7z7nhjdvbu/1.2.dressing_up_proteins.ppt?dl=0>Slides</a> <a href=https://www.dropbox.com/s/grokuesuq1icfyr/5.1.1.msa_overview.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Day 3</b></td><td>NORBIS</td><td>LECTURE</td><td><font size=+1><b>Multiple Sequence Alignment Algorithms</b></font></td><td> <a href=https://www.dropbox.com/s/mdj05j2bv7r4sho/1.0.BioinformaticsIntroduction.ppt?dl=0>Slides</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 3</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Parsing and manipulating binary trees</b></font></td><td><a href=./practicals/practicals_5/index.html>P</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 3</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Implementing your own multiple sequence aligner</b></font></td><td><a href=./practicals/practicals_6/index.html>P</a></td></tr><tr align=center BGCOLOR="#FFFFCC" ><td><b>Day 3</b></td><td>NORBIS</td><td>BIBLIO</td><td><font size=+1><b>MSA papers</b></font></td><td> <a href=>Biblio</a> <a href=https://www.dropbox.com/s/uj5cocko6clkcv8/chatzou2016.pdf?dl=0>Biblio</a> <a href=https://www.dropbox.com/s/st582pkoxwle19f/garriga2019.pdf?dl=0>Biblio</a> <a href=https://www.dropbox.com/s/q0oox4m6dp6tamq/notredame2000.pdf?dl=0>Biblio</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Day 4</b></td><td>NORBIS</td><td>LECTURE</td><td><font size=+1><b>RNA Folding Algorithms</b></font></td><td> <a href=https://www.dropbox.com/s/100na0f3uyg2dus/4.5.lecture.secondary_structure_prediction.ppt?dl=0>Slides</a> <a href=https://www.dropbox.com/s/0cq94s5ozxbbxkl/4.5.lecture.secondary_structure_prediction.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 4</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Adpating the Nussinv Algorithm</b></font></td><td><a href=./practicals/practicals_7/index.html>P</a></td></tr><tr align=center BGCOLOR="#FFFFCC" ><td><b>Day 4</b></td><td>NORBIS</td><td>BIBLIO</td><td><font size=+1><b>RNA Secondary Structure Prediction</b></font></td><td> <a href=https://www.dropbox.com/s/013v6wnng0xd9z6/4.5.paper.Eddy_sec_struc_primer.pdf?dl=0>Biblio</a> <a href=https://www.dropbox.com/s/gu2kt20u27mcu53/4.5.paper.RivasEddy_pseudo_knots.pdf?dl=0>Biblio</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b>Day 5</b></td><td>NORBIS</td><td>LECTURE</td><td><font size=+1><b>RNA Genomic Analysis using RNAseq</b></font></td><td> <a href=https://www.dropbox.com/s/5ewsch9qo5hhldx/4.4.lecture.short_read_mapping_algo.pptx?dl=0>Slides</a> <a href=https://www.dropbox.com/s/nye1sp3eikq43ks/4.4.lecture.short_read_mapping_algo.mp4?dl=0>Voiceover</a></td></tr><tr align=center BGCOLOR="#FFFFCC" ><td><b>Day 5</b></td><td>NORBIS</td><td>BIBLIO</td><td><font size=+1><b>Short Reads Mapping</b></font></td><td> <a href=https://www.dropbox.com/s/iptul04mgu13gvb/4.4.paper.gem.pdf?dl=0>Biblio</a> <a href=https://www.dropbox.com/s/z0g78qmv6ludefi/4.4.paper.StringMatchingBWT_ReadThroughTranscription.pdf?dl=0>Biblio</a> <a href=https://www.dropbox.com/s/734jx7x8q4x7j2z/kalisto2106.pdf?dl=0>Biblio</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr><tr align=center BGCOLOR="#FFFEAA" ><td><b>Day 5</b></td><td>NORBIS</td><td>PRACTICALS</td><td><font size=+1><b>Project Report - Delivery date: 15/06/2020</b></font></td><td><a href=./practicals/practicals_8/index.html>P</a></td></tr><tr align=center BGCOLOR="white" ><td colspan=6><font color="white">. </td></tr></table><BR><BR><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>REFERENCES</b></center><p align="justify"><p><align>1.        <a href="http://graveleylab.cam.uchc.edu/WebData/mduff/MEDS_6498_SPRING_2016/Durbin_Biological_Sequence_Analysis.pdf">Algorithms: Durbin et al., Biological Sequence Analysis, 1999, Oxford Press </A>
</p></align><p><align>2.        <a href="http://userpages.fu-berlin.de/digga/p4b.pdf">Algorithms: Python for Biologists: A complete programming course for beginners, Martin Jones,2013 , Createspace Independent Publishing Platform </a>
</p></align><p><align>3.        Evolution: Pathy, Protein Evolution, 2007, Blackwell
</p></align></p></td></tr></table><br><br><table BGCOLOR=white Align=center width=70% Border=10><tr><td><center><p><Small>This Entire Course Was Automatically Generated Using BED, the Bioinformatics Exercise Database. BED is a freeware available on request <a href="mailto:cedric.notredame@europe.com?subject=BED">Cedric Notredame</a></small></p></center></td></tr></table><BR><BR> </body> </html>