<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 9">
<meta name=Originator content="Microsoft Word 9">
<link rel=File-List href="./Realabstractweb_fichiers/filelist.xml">
<title>A new strategy for the design of functional signatures in protein
sequences</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Jean-michel CLAVERIE</o:Author>
  <o:Template>Normal</o:Template>
  <o:LastAuthor>Jean-michel CLAVERIE</o:LastAuthor>
  <o:Revision>2</o:Revision>
  <o:TotalTime>23</o:TotalTime>
  <o:Created>2001-06-19T08:58:00Z</o:Created>
  <o:LastSaved>2001-06-19T08:58:00Z</o:LastSaved>
  <o:Pages>1</o:Pages>
  <o:Company>cnrs</o:Company>
  <o:Lines>1</o:Lines>
  <o:Paragraphs>1</o:Paragraphs>
  <o:Version>9.3821</o:Version>
 </o:DocumentProperties>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:HyphenationZone>21</w:HyphenationZone>
 </w:WordDocument>
</xml><![endif]-->
<style>
<!--
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";
	mso-ansi-language:EN-US;}
p.MsoTitle, li.MsoTitle, div.MsoTitle
	{margin:0cm;
	margin-bottom:.0001pt;
	text-align:center;
	mso-pagination:widow-orphan;
	font-size:14.0pt;
	mso-bidi-font-size:12.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";
	mso-ansi-language:EN-GB;
	font-weight:bold;}
@page Section1
	{size:595.3pt 841.9pt;
	margin:70.85pt 70.85pt 70.85pt 70.85pt;
	mso-header-margin:35.4pt;
	mso-footer-margin:35.4pt;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
-->
</style>
</head>

<body lang=FR style='tab-interval:35.4pt'>

<div class=Section1>

<p class=MsoTitle><span lang=EN-GB>A new strategy for the design of functional
signatures in protein sequences</span></p>

<p class=MsoNormal align=center style='text-align:center'><b><span lang=EN-GB
style='font-size:14.0pt;mso-bidi-font-size:12.0pt;mso-ansi-language:EN-GB'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></b></p>

<p class=MsoNormal style='text-align:justify'><span lang=EN-GB
style='mso-ansi-language:EN-GB'>Regular expression-based motifs, at the origin
of the concept of functional signatures for proteins are progressively being
supplanted by more sophisticated probabilistic methods, such as position weight
matrices, Hidden Markov Models and Neural Networks. We will present a new
strategy to turn multiple alignments into specific regular expressions using an
interactive JAVA-based software system called REAL (Regular Expression Analysis
and Location). Our approach is based on the computation and parallel display of
various profiles computed both from the average value and variability, of
numerous amino acid properties. Relevant positions can also be selected on the
basis of their information content, relative to a given property. The
Information Content provides a quantitative measure of the contrast between
residue variability and property invariance. In the automatic mode, a simple
probabilistic framework is used to extract the most informative positions. Once
the position selected on the basis of one (or several) of the many available
criteria, a regular expression is automatically generated from the multiple
alignment using a simple quantitative rule. Tentative signatures are tested with
the companion database scanning program LOOKFOR. Possible applications of the
REAL software will be presented through two examples.<o:p></o:p></span></p>

</div>

</body>

</html>
