<HTML>
<HEAD>
   <META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
   <META NAME="generator" CONTENT="Adobe GoLive 4">
   <META NAME="GENERATOR" CONTENT="Mozilla/4.04 [en] (X11; I; IRIX 5.3 IP22) [Netscape]">
   <TITLE>Exercises for Day 4 of the EMBNET course</TITLE>
</HEAD>
<BODY TEXT="#000000" BGCOLOR="#F0FFFF" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000">

<CENTER>
<H1>
<B><FONT SIZE=+1></FONT></B></H1></CENTER>
<B><FONT SIZE=+2>Extraction of a protein domain</FONT></B><BR>

<HR width="100%">
<HR width="100%">
<H2>
<A NAME="MAPP.4"></A>Finding a local homology domain by BLAST and pairwise
alignment</H2>

<BLOCKQUOTE><B>a)</B> Find the sequence for the human DNA repair gene Xrcc1
in SwissProt and create a local copy.

<P><B>b)</B> Perform a <B><I>blast</I></B> search against the non redundant
protein database. <B>**NOTE**:</B> this sequence contains low complexity
regions, therefore use <I><B>blast</B>'s</I> filter option, otherwise lots
of junk will be reported

<P><B>c)</B> By using pairwise alignment, check which of the matches are
significant and which are potentially interesting. Check the high scoring
matches for putative biological relevance and note the results.

<P><B>d)</B> Find the conserved region of Xrcc1 and Rad4 by using dotplots.
(Use Window=35, Stringency=20) Note the internal repeat in Rad4.</BLOCKQUOTE>
<A HREF="msa_2_sol.html#MAPP.4">Hints and Solutions:</A>

<P>
<HR width="100%">
<HR width="100%">
<H2>
<A NAME="MAPP.5"></A>Multiple alignment of homology domains an profile
searches</H2>

<BLOCKQUOTE>a) use <B><I>pileup</I></B> to create a multiple alignment
of these three sequence fragments (in MSF format)

<P>b) use <B><I>lineup</I></B> to look at the multiple alignment and to
remove potential non-conserved overhanging ends. Save the edited multiple
alignment in MSF format.

<P>c) use <B><I>profilemake</I></B> to create a profile from the edited
MSF file. Create the profile with lineas symbol weighting and stringent
treatment of non-observed symbols.

<P>d) use <B><I>profilesearch</I></B> to search the profile against a protein
sequence database. Run this search with compositional averaging switched
off. Don't forget to restrict the output list to the 100 best sequences,
otherwise ALL SEQUENCES in the database will be reported ! Profile searches
against big databases take a lot of time and cpu resources. The output
file of a <B><I>profilesearch</I></B> against the complete SwissProt database
will be provided.
<BR>NOTE: use the substitution matrix blossum45.cmp&nbsp; with the option
-data=genmoredata:blosum45.cmp)

<P>e)<FONT SIZE=+0> Check the output list for high scoring matches of potential
relevance. Create profile-to-sequence alignments with selected examples
using profilesegments or profilegap.</FONT></BLOCKQUOTE>

<H2>
<FONT SIZE=+0>Advanced exercises</FONT></H2>

<BLOCKQUOTE>The following exercises might be too time consuming to perform
now -maybe try them at home!

<P>f) Use <B><I>profilegap</I></B> to extract from the high-scoring sequences
those fragments that match with the profile.

<P>g) Add these new sequence segments to the existing multiple alignment,
either by <B><I>pileup</I></B> or by manual editing.

<P>h) Repeat the <B><I>profilemake</I></B>/<B><I>profilesearch</I></B>
cycle iteratively as described above.</BLOCKQUOTE>
<A HREF="msa_2_sol.html">Hints and Solutions:</A>
<BR><A HREF="exercises.html">[BACK]</A>
</BODY>
</HTML>
