<html><meta http-equiv="content-type" content="text/html;charset=iso-8859-1"><title>An introduction to Protein Sequence Alignments<p>Tangier, June 2011</p></title></head><BODY TEXT="#000000" BGCOLOR="#FFEEDD" LINK="#0000EE" VLINK="#551A8B" ALINK="#FF0000"><center><table border=10 cellpadding="20" width="580" nosave><tr> <td align=center bgcolor="#CC9933"><h2><font color=white>An introduction to Protein Sequence Alignments<p>Tangier, June 2011</p><p>Cedric Notredame</p></font></h2></td></tr></table><br><br><br><center><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>OBJECTIVES</b></center><p align="justify">The goal of this course is to introduce various aspects of protein sequence alignments. We will start introducing the notion of substitution matrix, insisting on the necessary biological relevance of the considered models. We will then introduce Blast, the most commonly used application of pairwise sequence alignments. In the second part of the course, we will introduce the notion of multiple sequence alignments. We will develop the strength and weaknesses of the most commo widely used strategies like ClustalW. The last part of the course will introduce state of the art methods for the comparison of protein and RNA sequences, including T-Coffee, a software developped in my lab.
</p></td></tr></table><br><br><h1><font size="2">Send your Questions to: <a href="mailto:cedric.notredame@crg.eu">cedric.notredame@crg.eu</a></h1></font><br><br><table BGCOLOR="#CC9966"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr align=center BGCOLOR="#CC9933" ><td><b><font size=+2 color=white>Date</font></b></td><td><b><font size=+2 color=white>Location</font></b></td><td><b><font size=+2 color=white>Session</font></b></td><td><b><font size=+2 color=white>Title</font></b></td><td><b><font size=+2 color=white>Links</font></b></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> 1 </b></td><td>---</td><td>LECTURE</td><td><font size=+1><b>Introduction Protein Substitution Matrices</b></font></td><td><a href=./lectures/4.1.pairwise_intro.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> 2 </b></td><td>---</td><td>LECTURE</td><td><font size=+1><b>Searching Protein Databases with Blast</b></font></td><td><a href=./lectures/4.2.pairwise_blast.ppt>L</a></td></tr><tr align=center BGCOLOR="#99FFCC" ><td><b> 3 </b></td><td>---</td><td>LECTURE</td><td><font size=+1><b>Introduction to Multiple Sequence Alignments</b></font></td><td><a href=./lectures/5.1.msa_introduction.ppt>L</a></td></tr><tr align=center BGCOLOR="#CCFFFF" ><td><b> 4 </b></td><td>---</td><td>SEMINAR</td><td><font size=+1><b>Protein and RNA multiple Sequence Alignments with the T-Coffee package</b></font></td><td><a href=./seminars/T_coffee_Tangier_2011.ppt>S</a></td></tr></table><BR><BR><br><table BGCOLOR="white"Align="center" cellpadding="30"  leftmargin="30" marginwidth="30" width="70%"  border="5"><tr BGCOLOR="white" ><td colspan=6><b><center>REFERENCES</b></center><p align="justify"><p><align>1.	Durbin, R., et al., Biological Sequence Analysis. 1998, Cambridge: Cambridge University Press.
</p></align><p><align>2.	Altschul, S.F., et al., Gapped BLAST and PSI-BLAST: a new generation of protein database search programs. nucleic acids res., 1997. 25(17): p. 2289-3402.
</p></align><p><align>3.	Notredame, C., D.G. Higgins, and J. Heringa, T-Coffee: A novel method for fast and accurate multiple sequence alignment. J Mol Biol, 2000. 302(1): p. 205-17.
</p></align><p><align>4.	Wallace, I.M., et al., M-Coffee: combining multiple sequence alignment methods with T-Coffee. Nucleic Acids Res, 2006. 34(6): p. 1692-9.
</p></align><p><align>5.	Wilm, A., D.G. Higgins, and C. Notredame, R-Coffee: a method for multiple alignment of non-coding RNA. Nucleic Acids Res, 2008. 36(9): p. e52.
</p></align><p><align>6.	Bussoti, G., et al., BlastR: A new Tool for searching RNA in databases. Nucleic Acids Research, 2011.
</p></align></p></td></tr></table><br><br><table BGCOLOR=white Align=center width=70% Border=10><tr><td><center><p><Small>This Entire Course Was Automatically Generated Using BED, the Bioinformatics Exercise Database. BED is a freeware available on request <a href="mailto:cedric.notredame@europe.com?subject=BED">Cedric Notredame</a></small></p></center></td></tr></table><BR><BR> </body> </html>